src/charm.py

```diff
@@ -437,7 +493,10 @@ def _on_peer_relation_changed(self, event: HookEvent) -> None:
             event.defer()
             return
 
-        self.postgresql_client_relation.update_read_only_endpoint()
+        try:
+            self.postgresql_client_relation.update_read_only_endpoint()
+        except ModelError as e:
+            logger.debug("Cannot update read_only endpoints: %s", str(e))
```

Comment:
This should most probably be logged as error.

Reply:
I thought about this, but AFAICT this error occurring (and getting caught here) did not immediately affected the charm's behavior, and the user wouldn't be able to do anything about it (charm doesn't get blocked or anything), so would it be useful to have it as a big red error log? @marceloneppel what do you think?

I don't have a strong opinion on this, just trying to better understand what it takes for a log to be high level (warning/error)

Reply:
A warning is better as it didn't affect the charm behavior, but may eventually lead to undesirable effects later.

Reply:
Warning it is!